### PR TITLE
Add advertise-status-addr for tiflash proxy

### PR DESCRIPTION
**Pull request by birdstorm**
### What problem does this PR solve? 

Add `advertise-status-addr` for tiflash to support host name.

### Check List 

Tests 

 - No code

Release notes:

```release-note
Add `advertise-status-addr` for tiflash to support host name
```

### CODE REVIEW

**Comment by birdstorm** [comment]
@solotzg PTAL

File: components/playground/instance/tiflash_proxy_config.go [hunk 1]
```diff
@@ -43,11 +46,42 @@ data-dir = "%[6]s"
...
+[security]
+ca-path = ""
+cert-path = ""
+key-path = ""
+
+[server]
+addr = "0.0.0.0:%[4]d"
+advertise-addr = "%[2]s:%[4]d"
+engine-addr = "%[2]s:%[3]d"
+status-addr = "0.0.0.0:%[5]d"
```
Comment (solotzg) [inline components/playground/instance/tiflash_proxy_config.go:65]: Maybe we can make some abstraction and reduce similar code

File: pkg/cluster/spec/tiflash.go [hunk 2]
```diff
@@ -241,19 +242,26 @@ server_configs:
...
 
 // InitTiFlashLearnerConfig initializes TiFlash learner config file
-func (i *TiFlashInstance) InitTiFlashLearnerConfig(cfg *scripts.TiFlashScript, src map[string]interface{}) (map[string]interface{}, error) {
+func (i *TiFlashInstance) InitTiFlashLearnerConfig(cfg *scripts.TiFlashScript, clusterVersion string, src map[string]interface{}) (map[string]interface{}, error) {
 	topo := Specification{}
+	var statusAddr string
 
 	firstDataDir := strings.Split(cfg.DataDir, ",")[0]
 
+	if semver.Compare("v4.0.5", clusterVersion) <= 0 {
```
Comment (lonng) [inline pkg/cluster/spec/tiflash.go:251]: Prefer 
```suggestion
	if semver.Compare(clusterVersion, "v4.0.5") >= 0 {
```

Comment (july2993) [reply pkg/cluster/spec/tiflash.go:251]: > An invalid semantic version string is considered less than a valid one.

And "nightly" is an invalid one, do we need to handle it specifically?

Comment (birdstorm) [reply pkg/cluster/spec/tiflash.go:251]: nightly version is not affected. This config change is usable since v4.0.4, so is nightly.

**Comment by birdstorm** [comment]
@lonng PTAL again, thanks.

**Pull request merged**